estate/models/estate_property.py

```diff
@@ -0,0 +1,150 @@
+    )
+
+    name = fields.Char('Name', required=True, default='My new house')
+    description = fields.Text('Description')
+    active = fields.Boolean(default=True)
+    postcode = fields.Char('Postcode')
+    date_availability = fields.Date('Available From', default=lambda self: fields.Date.context_today(self) + timedelta(days=90), copy=False)
+    expected_price = fields.Float('Expected Price', required=True)
+    selling_price = fields.Float('Selling Price', readonly=True, copy=False)
+    bedrooms = fields.Integer('Bedrooms', default=2)
+    living_area = fields.Integer('Living Area (sqm)')
+    facades = fields.Integer('#Facades')
+    garage = fields.Boolean('Garage')
+    garden = fields.Boolean('Garden')
+    garden_area = fields.Integer('Garden area')
+
+    garden_orientation = fields.Selection([
+        ("north", "North"),
+        ("south", "South"),
+        ("east", "East"),
+        ("west", "West"),
+    ])
+
+    state = fields.Selection(
+        [
+            ('new', 'New'),
+            ('offer_received', 'Offer Received'),
+            ('offer_accepted', 'Offer Accepted'),
+            ('sold', 'Sold'),
+            ('cancelled', 'Cancelled'),
+        ],
+        required=True,
+        copy=False,
+        default='new',
+    )
+    property_type_id = fields.Many2one(
+        "estate.property.type",
+        string="Property Type"
+    )
+
+    buyer_id = fields.Many2one(
+        "res.partner",
+        string="Buyer",
+        copy=False
+    )
+
+    salesperson_id = fields.Many2one(
+        "res.users",
+        string="Salesman",
+        default=lambda self: self.env.user
+    )
+
+    tag_ids = fields.Many2many(
+        "estate.property.tag",
+        string="Tags"
+    )
+
+    offer_ids = fields.One2many(
+        "estate.property.offer",
+        "property_id",
+        string="Offers",
+    )
+
+    total_area = fields.Float(
+        compute="_compute_total_area",
+        string="Total Area",
+    )
+
+    best_price = fields.Float(
+        compute="_compute_best_price",
+        string="Best Offer",
+    )
+
+    @api.depends("living_area", "garden_area")
+    def _compute_total_area(self):
+        for record in self:
+            record.total_area = record.living_area + record.garden_area
+
+    @api.depends("offer_ids.price")
+    def _compute_best_price(self):
+        for record in self:
+            if record.offer_ids:
+                record.best_price = max(record.offer_ids.mapped("price"))
+            else:
+                record.best_price = 0.0
+
+    @api.onchange("garden")
+    def _onchange_garden(self):
+        if self.garden:
+            self.garden_area = 10
+            self.garden_orientation = "north"
+        else:
+            self.garden_area = 0
+            self.garden_orientation = False
+
+    def action_cancel(self):
+        for record in self:
+            if record.state == "sold":
+                raise UserError("A sold property cannot be cancelled.")
```

Comment:
```suggestion
                raise UserError(self.env._("A sold property cannot be cancelled."))
```
In Odoo, user-facing strings outside the database must be wrapped in `_()` to enable translation.

(I'm only putting the comment here, but this applies to various places in this PR)